### PR TITLE
Replace SpringExtension with MockitoExtension

### PR DESCRIPTION
**Pull request by YiannisPapastergiou**
Replace SpringExtension with MockitoExtension in mapper and model validation unit test classes

### JIRA link
Resolves: [BI-8683](https://companieshouse.atlassian.net/browse/BI-8683)

### Change description
Replace SpringExtension with MockitoExtension in mapper and model validation unit test classes

### Work checklist

* [ ] Bug fix
* [ ] New feature
* [ ] Breaking change

### Pull request checklist

* [ ] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [X] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/mapper/AppealMapperTest.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
 import uk.gov.companieshouse.model.PenaltyIdentifier;
 import uk.gov.companieshouse.model.Reason;
 
-@ExtendWith(SpringExtension.class)
+@ExtendWith(MockitoExtension.class)
 public class AppealMapperTest {
```
Comment (mreis-ch) [inline src/test/java/uk/gov/companieshouse/mapper/AppealMapperTest.java:26]: Remove **public** from class declaration. Also, we use the **@DisplayName** annotation on test methods to give a more verbose way of describing a test. Can this annotation be added to test methods in the class please.

File: src/test/java/uk/gov/companieshouse/mapper/AttachmentMapperTest.java [hunk 2]
```diff
@@ -6,13 +6,13 @@
...
-import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.mockito.junit.jupiter.MockitoExtension;
 import uk.gov.companieshouse.TestData;
 import uk.gov.companieshouse.database.entity.AttachmentEntity;
 import uk.gov.companieshouse.model.Attachment;
 import uk.gov.companieshouse.TestUtil;
 
-@ExtendWith(SpringExtension.class)
+@ExtendWith(MockitoExtension.class)
 public class AttachmentMapperTest {
```
Comment (mreis-ch) [inline src/test/java/uk/gov/companieshouse/mapper/AttachmentMapperTest.java:16]: Remove **public** from class declaration. Also, we use the **@DisplayName** annotation on test methods to give a more verbose way of describing a test. Can this annotation be added to test methods in the class please.

File: src/test/java/uk/gov/companieshouse/mapper/CreatedByMapperTest.java [hunk 3]
```diff
@@ -6,13 +6,13 @@
...
-import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.mockito.junit.jupiter.MockitoExtension;
 import uk.gov.companieshouse.TestData;
 import uk.gov.companieshouse.TestUtil;
 import uk.gov.companieshouse.database.entity.CreatedByEntity;
 import uk.gov.companieshouse.model.CreatedBy;
 
-@ExtendWith(SpringExtension.class)
+@ExtendWith(MockitoExtension.class)
 public class CreatedByMapperTest {
```
Comment (mreis-ch) [inline src/test/java/uk/gov/companieshouse/mapper/CreatedByMapperTest.java:16]: Remove **public** from class declaration. Also, we use the **@DisplayName** annotation on test methods to give a more verbose way of describing a test. Can this annotation be added to test methods in the class please.

File: src/test/java/uk/gov/companieshouse/mapper/IllnessReasonMapperTest.java [hunk 4]
```diff
@@ -7,14 +7,14 @@
...
+import org.mockito.junit.jupiter.MockitoExtension;
 import uk.gov.companieshouse.TestData;
 import uk.gov.companieshouse.database.entity.IllnessReasonEntity;
 import uk.gov.companieshouse.model.IllnessReason;
 import uk.gov.companieshouse.TestUtil;
 
 
-@ExtendWith(SpringExtension.class)
+@ExtendWith(MockitoExtension.class)
 class IllnessReasonMapperTest {
```
Comment (mreis-ch) [inline src/test/java/uk/gov/companieshouse/mapper/IllnessReasonMapperTest.java:18]: We use the **@DisplayName** annotation on test methods to give a more verbose way of describing a test. Can this annotation be added to test methods in the class please.

File: src/test/java/uk/gov/companieshouse/mapper/OtherReasonMapperTest.java [hunk 5]
```diff
@@ -10,15 +10,15 @@
...
 import uk.gov.companieshouse.TestData;
 import uk.gov.companieshouse.database.entity.AttachmentEntity;
 import uk.gov.companieshouse.database.entity.OtherReasonEntity;
 import uk.gov.companieshouse.model.Attachment;
 import uk.gov.companieshouse.model.OtherReason;
 import uk.gov.companieshouse.TestUtil;
 
-@ExtendWith(SpringExtension.class)
+@ExtendWith(MockitoExtension.class)
 public class OtherReasonMapperTest {
```
Comment (mreis-ch) [inline src/test/java/uk/gov/companieshouse/mapper/OtherReasonMapperTest.java:22]: Remove **public** from class declaration. Also, we use the **@DisplayName** annotation on test methods to give a more verbose way of describing a test. Can this annotation be added to test methods in the class please.

File: src/test/java/uk/gov/companieshouse/mapper/PenaltyIdentifierMapperTest.java [hunk 6]
```diff
@@ -6,12 +6,12 @@
...
 import org.junit.jupiter.api.extension.ExtendWith;
-import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.mockito.junit.jupiter.MockitoExtension;
 import uk.gov.companieshouse.TestData;
 import uk.gov.companieshouse.database.entity.PenaltyIdentifierEntity;
 import uk.gov.companieshouse.model.PenaltyIdentifier;
 
-@ExtendWith(SpringExtension.class)
+@ExtendWith(MockitoExtension.class)
 public class PenaltyIdentifierMapperTest {
```
Comment (mreis-ch) [inline src/test/java/uk/gov/companieshouse/mapper/PenaltyIdentifierMapperTest.java:15]: Remove **public** from class declaration. Also, we use the **@DisplayName** annotation on test methods to give a more verbose way of describing a test. Can this annotation be added to test methods in the class please.

File: src/test/java/uk/gov/companieshouse/mapper/ReasonMapperTest.java [hunk 7]
```diff
@@ -11,15 +11,15 @@
...
 import uk.gov.companieshouse.database.entity.IllnessReasonEntity;
 import uk.gov.companieshouse.database.entity.OtherReasonEntity;
 import uk.gov.companieshouse.database.entity.ReasonEntity;
 import uk.gov.companieshouse.model.IllnessReason;
 import uk.gov.companieshouse.model.OtherReason;
 import uk.gov.companieshouse.model.Reason;
 
-@ExtendWith(SpringExtension.class)
+@ExtendWith(MockitoExtension.class)
 class ReasonMapperTest {
```
Comment (mreis-ch) [inline src/test/java/uk/gov/companieshouse/mapper/ReasonMapperTest.java:23]: We use the **@DisplayName** annotation on test methods to give a more verbose way of describing a test. Can this annotation be added to test methods in the class please.

File: src/test/java/uk/gov/companieshouse/model/validator/AppealReasonValidatorTest.java [hunk 8]
```diff
@@ -10,10 +10,10 @@
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.InjectMocks;
 
-import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.mockito.junit.jupiter.MockitoExtension;
 import uk.gov.companieshouse.model.Reason;
 
-@ExtendWith(SpringExtension.class)
+@ExtendWith(MockitoExtension.class)
 class AppealReasonValidatorTest {
```
Comment (mreis-ch) [inline src/test/java/uk/gov/companieshouse/model/validator/AppealReasonValidatorTest.java:17]: We use the **@DisplayName** annotation on test methods to give a more verbose way of describing a test. Can this annotation be added to test methods in the class please.

File: src/test/java/uk/gov/companieshouse/model/validator/CompanyNumberValidatorTest.java [hunk 9]
```diff
@@ -11,10 +11,7 @@
 import java.util.stream.Stream;
 
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.extension.ExtendWith;
-import org.springframework.test.context.junit.jupiter.SpringExtension;
 
-@ExtendWith(SpringExtension.class)
 public class CompanyNumberValidatorTest {
```
Comment (mreis-ch) [inline src/test/java/uk/gov/companieshouse/model/validator/CompanyNumberValidatorTest.java:15]: Remove **public** from class declaration. Also, we use the **@DisplayName** annotation on test methods to give a more verbose way of describing a test. Can this annotation be added to test methods in the class please.

File: src/test/java/uk/gov/companieshouse/model/validator/RelationshipValidatorTest.java [hunk 10]
```diff
@@ -6,14 +6,14 @@
...
+import org.mockito.junit.jupiter.MockitoExtension;
 import uk.gov.companieshouse.TestData;
 import uk.gov.companieshouse.TestUtil;
 import uk.gov.companieshouse.model.Appeal;
 import uk.gov.companieshouse.model.CreatedBy;
 import uk.gov.companieshouse.model.Reason;
 
-@ExtendWith(SpringExtension.class)
+@ExtendWith(MockitoExtension.class)
 class RelationshipValidatorTest {
```
Comment (mreis-ch) [inline src/test/java/uk/gov/companieshouse/model/validator/RelationshipValidatorTest.java:17]: We use the **@DisplayName** annotation on test methods to give a more verbose way of describing a test. Can this annotation be added to test methods in the class please.

File: src/test/java/uk/gov/companieshouse/model/validator/AppealReasonValidatorTest.java [hunk 11]
```diff
@@ -10,10 +10,10 @@
...
 
     @InjectMocks
```
Comment (CH-JonathanBlake) [inline src/test/java/uk/gov/companieshouse/model/validator/AppealReasonValidatorTest.java:19]: No mocks to inject.

**Review by mreis-ch — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/companieshouse/mapper/IllnessReasonMapperTest.java (hunk 4 above)
Comment (YiannisPapastergiou) [reply src/test/java/uk/gov/companieshouse/mapper/IllnessReasonMapperTest.java:19]: Done in new commit

File: src/test/java/uk/gov/companieshouse/model/validator/AppealReasonValidatorTest.java (hunk 8 above)
Comment (YiannisPapastergiou) [reply src/test/java/uk/gov/companieshouse/model/validator/AppealReasonValidatorTest.java:12]: Done in new commit

File: src/test/java/uk/gov/companieshouse/model/validator/RelationshipValidatorTest.java (hunk 10 above)
Comment (YiannisPapastergiou) [reply src/test/java/uk/gov/companieshouse/model/validator/RelationshipValidatorTest.java:15]: Done in new commit

File: src/test/java/uk/gov/companieshouse/model/validator/AppealReasonValidatorTest.java (hunk 11 above)
Comment (YiannisPapastergiou) [reply src/test/java/uk/gov/companieshouse/model/validator/AppealReasonValidatorTest.java]: refactored in new commit

File: src/test/java/uk/gov/companieshouse/mapper/AttachmentMapperTest.java (hunk 2 above)
Comment (YiannisPapastergiou) [reply src/test/java/uk/gov/companieshouse/mapper/AttachmentMapperTest.java]: Done in new commit

File: src/test/java/uk/gov/companieshouse/mapper/AppealMapperTest.java (hunk 1 above)
Comment (YiannisPapastergiou) [reply src/test/java/uk/gov/companieshouse/mapper/AppealMapperTest.java]: Done in new commit

File: src/test/java/uk/gov/companieshouse/mapper/OtherReasonMapperTest.java (hunk 5 above)
Comment (YiannisPapastergiou) [reply src/test/java/uk/gov/companieshouse/mapper/OtherReasonMapperTest.java]: Done in new commit

File: src/test/java/uk/gov/companieshouse/mapper/ReasonMapperTest.java (hunk 7 above)
Comment (YiannisPapastergiou) [reply src/test/java/uk/gov/companieshouse/mapper/ReasonMapperTest.java:24]: Done in new commit

File: src/test/java/uk/gov/companieshouse/model/validator/CompanyNumberValidatorTest.java (hunk 9 above)
Comment (YiannisPapastergiou) [reply src/test/java/uk/gov/companieshouse/model/validator/CompanyNumberValidatorTest.java]: Done in new commit

File: src/test/java/uk/gov/companieshouse/model/validator/AppealReasonValidatorTest.java (hunk 11 above)
Comment (mreis-ch) [reply src/test/java/uk/gov/companieshouse/model/validator/AppealReasonValidatorTest.java]: The **@InjectMocks** is still there. As Jonathan said, this needs to be removed as you're mocking the class you want to test so you're sort of not testing it when it should be a real instance. Either instantiate the private AppealReasonValidator object at declaration or probably better to have a setup method which instantiates before each test like this:

    @BeforeEach
    private void setup() {
    	appealReasonValidator = new AppealReasonValidator();
    }

**Review by mreis-ch — CHANGES_REQUESTED** [review]
A couple of comments on the AppealReasonValidator.java

File: src/test/java/uk/gov/companieshouse/model/validator/AppealReasonValidatorTest.java [hunk 12]
```diff
@@ -10,10 +10,10 @@
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.InjectMocks;
 
-import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.mockito.junit.jupiter.MockitoExtension;
 import uk.gov.companieshouse.model.Reason;
 
-@ExtendWith(SpringExtension.class)
+@ExtendWith(MockitoExtension.class)
```
[the remark this answers is not in the document]
Comment (YiannisPapastergiou) [reply src/test/java/uk/gov/companieshouse/model/validator/AppealReasonValidatorTest.java]: This comment is on an outdated version so I will resolve it 👍

File: src/test/java/uk/gov/companieshouse/model/validator/AppealReasonValidatorTest.java (hunk 11 above)
Comment (YiannisPapastergiou) [reply src/test/java/uk/gov/companieshouse/model/validator/AppealReasonValidatorTest.java]: New commit resolving this pushed

**Review by mreis-ch — APPROVED** [review]
Looks good 👍

**Pull request merged**